pkg/auth/ldaputil/ldapclient/interfaces.go

```diff
@@ -0,0 +1,32 @@
+package ldapclient
+
+import (
+	"crypto/tls"
+
+	"github.com/go-ldap/ldap"
+)
+
+// Client knows how to interact with an LDAP server
+type Client interface {
+	Start()
+	StartTLS(config *tls.Config) error
+	Close()
+
+	Bind(username, password string) error
+	SimpleBind(simpleBindRequest *ldap.SimpleBindRequest) (*ldap.SimpleBindResult, error)
+
+	Modify(modifyRequest *ldap.ModifyRequest) error
+
+	Compare(dn, attribute, value string) (bool, error)
+	PasswordModify(passwordModifyRequest *ldap.PasswordModifyRequest) (*ldap.PasswordModifyResult, error)
+
+	Search(searchRequest *ldap.SearchRequest) (*ldap.SearchResult, error)
+	SearchWithPaging(searchRequest *ldap.SearchRequest, pagingSize uint32) (*ldap.SearchResult, error)
+}
+
+// Config knows how to connect to an LDAP server and can describe which server it is connecting to
+type Config interface {
+	Connect() (client Client, err error)
+	GetBindCredentials() (bindDN, bindPassword string)
+	DescribeLDAPURL() (schemehostport string)
```

Comment:
> This is used in ldaputil somewhere for an error message... Would be more opaque as String() since it's asking for some subset of information from the config.

Too icky to keep.